events/repostream.go

```diff
@@ -13,12 +13,38 @@ import (
 
 type LiteStreamHandleFunc func(op repomgr.EventKind, seq int64, path string, did string, rcid *cid.Cid, rec any) error
 
-func ConsumeRepoStreamLite(ctx context.Context, con *websocket.Conn, cb LiteStreamHandleFunc) error {
+type GetRepoFunc func(ctx context.Context, did string, oldest *string, newest string) ([]byte, error)
```

Comment:
I'd be curious to see an example implementation of this signature. Seems like it would want a reference to a cache of DID lookups (`did:plc` or `did:web`), or at least a DID PLC API client.

Not a blocking thing, we can add those optimizations later.